src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -83,6 +88,37 @@ public TypeFactory(OutputLibrary library)
             _ => throw new Exception("Unknown type")
         };
 
+        /// <summary>
+        /// This function is used to create a CSharpType from an InputType.
+        /// If the InputType is a framework type, the CSharpType will be created with the literal value Constant
+        /// object.
+        /// </summary>
+        /// <param name="inputType">The input type to create a CSharpType from.</param>
+        /// <param name="literalValue">The literal value of the InputType, if any.</param>
+        /// <param name="isLiteral">Flag that is used to determine if the constructed type is from a literal type.</param>
+        /// <returns></returns>
+        public CSharpType CreateType(InputType inputType, object literalValue, bool isLiteral)
```

Comment:
This logic seems it should be inside CSharpType itself.  Perhaps in the ctor or a static method on CsharpType called `FromLiteral`